internal/command/machine/clone.go

```diff
@@ -159,15 +160,34 @@ func runMachineClone(ctx context.Context) (err error) {
 		fmt.Fprintf(io.Out, "Auto destroy enabled and will destroy machine on exit. Use --clear-auto-destroy to remove this setting.\n")
 	}
 
-	if volID := flag.GetString(ctx, "attach-volume"); volID != "" {
-		if len(source.Config.Mounts) != 1 {
-			return fmt.Errorf("Can't attach the volume as the source machine doesn't have any volumes configured")
+	var volID string
+	if volumeInfo := flag.GetString(ctx, "attach-volume"); volumeInfo != "" {
+		splitVolumeInfo := strings.Split(volumeInfo, ":")
+
+		if len(source.Config.Mounts) > 1 {
+			return fmt.Errorf("Can't use --attach-volume for machines with more than 1 volume.")
+		} else if len(source.Config.Mounts) == 1 && len(splitVolumeInfo) > 1 {
+			return fmt.Errorf("Can't set a mount path on a machine with a volume, please use only the volume id on '%s'", volumeInfo)
```

Comment:
In the case the mount exists, It is possible to change both: the volume and the path when launching a new machine. there is nothing wrong with it.

Reply:
yeap-- maybe in a follow up PR?

I thought this is a very specific edge-case that shouldn't prevent anything meaningful. At the same time, it could potentially prevent some errors if the user mistypes the mount path.

Reply:
there is nothing wrong with changing the mount path and `fly machine` are low level surgical commands the user must be aware of. 

Even `fly deploy` allows changing the mount path, it prints a warning message, that is all.

Reply:
> Even `fly deploy` allows changing the mount path, it prints a warning message, that is all.

fair enough, didn't know about it-- and it was simple enough to change it.

can you please take another look? (I've tested quite a few cases locally, but it's my first PR on `flyctl` and the first time something I wrote in `go` goes live hehe)

Reply:
also, I didn't quite like changing values on `source.Config.Mounts` but as far as I could tell it's the only way w/o a big-ish refactor